app/instrumentation-tests/src/org/commcare/androidTests/DateWidgetsTests.kt

```diff
@@ -0,0 +1,171 @@
+            Collections.rotate(list,-index)
+
+            //asserts if months are present in order
+            for (listItem in list){
+                assertTrue(onView(withText(listItem)).isPresent())
+                onView(withId(R.id.monthupbtn)).perform(ViewActions.click())
+            }
+
+            onView(withId(R.id.yeardownbtn)).perform(ViewActions.click())
+            val date_selected = setDateToUniversalCalender(-10)
+            val gregorian_date = InstrumentationUtility.getText(onView(withId(R.id.dateGregorian)))
+            val formatted_date = formatGregorianDate(gregorian_date.drop(1).dropLast(1))
+            InstrumentationUtility.nextPage()
+            assertTrue(onView(withSubstring(formatted_date)).isPresent())
+            assertTrue(onView(withSubstring(date_selected)).isPresent())
+            InstrumentationUtility.submitForm()
+            assertTrue(onView(ViewMatchers.withText("1 form sent to server!")).isPresent())
+        }
+        InstrumentationUtility.logout()
+    }
+
+
+    @Test
+    fun testStandardWidget(){
+        InstrumentationUtility.openModule(4)
+        InstrumentationUtility.rotateLeft()
+        val date = setDateTo(-10)
+        val dateFormates = getDateInDifferentFormats(date)
+        InstrumentationUtility.rotatePortrait()
+        InstrumentationUtility.nextPage()
+        for (formats in dateFormates){
+            assertTrue(onView(allOf(withSubstring(formats))).isPresent())
+        }
+        InstrumentationUtility.submitForm()
+        assertTrue(onView(ViewMatchers.withText("1 form sent to server!")).isPresent())
+        InstrumentationUtility.logout()
+    }
+
+    /**
+     * function to set date in the date picker
+     */
+    fun setDateTo(days: Int) : Date {
+        val calendar = Calendar.getInstance()
+        calendar.add(Calendar.DATE, +days)
+
+        val year: Int = calendar.get(Calendar.YEAR)
+        val month: Int = calendar.get(Calendar.MONTH) + 1
+        val day: Int = calendar.get(Calendar.DAY_OF_MONTH)
+        onView(
+            ViewMatchers.withClassName(
+                Matchers.equalTo(
+                    DatePicker::class.java.name
+                )
+            )
+        ).perform(PickerActions.setDate(year, month, day))
+
+        return calendar.time
+    }
+
+    /**
+     * function returns different format of the selected date
+     */
+    fun getDateInDifferentFormats(date : Date) : Array<String>{
+        val long_format = SimpleDateFormat("EE, MMM dd, yyyy").format(date)
+        val short_format = SimpleDateFormat("d/M/yy").format(date)
+        val unformatted_date = SimpleDateFormat("yyyy-MM-dd").format(date)
+        return arrayOf(long_format.toString(),short_format.toString(),unformatted_date.toString())
+    }
+    /**
+     * function to set date in the date universal picker
+     */
+    fun setDateToUniversalCalender(days: Int) : String{
+
+        for (day in 1..days.absoluteValue){
+            if (days > 0) {
```

Comment:
you can move the if check outside the loop in order to not do it repeatedly - 
````
var dayChangeViewInteraction = if(days > 0) onView(withId(R.id.dayupbtn)) else onView(withId(R.id. daydownbtn))
for (day in 1..days.absoluteValue){
dayChangeViewInteraction.perform(ViewActions.click())
}
````